File: micropython/benchmarks/sum.py
SCALE_FACTOR = 100  # Define SCALE_FACTOR, adjust as needed


def benchmark():
    sum_value = 0
    for i in range(SCALE_FACTOR + 1):
        sum_value += i

    expected_sum = (SCALE_FACTOR * (SCALE_FACTOR + 1)) // 2

    return sum_value == expected_sum

File: native/benchmarks/sum.c
#ifndef SCALE_FACTOR
#define SCALE_FACTOR 1
#endif

int __attribute__ ((noinline))
benchmark()
{
    int volatile sum = 0;
    for (int i = 0; i <= SCALE_FACTOR; i++) {
        sum += i;
    }

    int expected_sum = (SCALE_FACTOR * (SCALE_FACTOR + 1)) / 2;
    return sum == expected_sum;
}